src/mesh/dim2/adjacency_map/adjacency_map.cpp

```diff
@@ -0,0 +1,472 @@
+#include "enumerations/dimension.hpp"
+#include "enumerations/specfem_enums.hpp"
+#include "mesh/dim2/mesh.hpp"
+
+#include "index_mapping_from_adjacencies.cpp"
+#include <list>
+#include <stdexcept>
+#include <utility>
+
+// ====================================================================
+// conversions between edge and boundary enums.
+
+static inline specfem::enums::edge::type
+bdtype_to_edge(const specfem::enums::boundaries::type &bd) {
```

Comment:
If you could define the correct numbers in `include/enumerations/specfem_enums.hpp` for edge and boundary, functions `detype_to_edge` and `edge_to_bdtype` can be simplified.